F-FrontEnd/src/F-compile-decl.c

```diff
@@ -2301,11 +2301,15 @@ declare_id_type(ID id, TYPE_DESC tp)
             tpp = tp;
             while(TYPE_REF(tpp) != NULL && IS_ARRAY_TYPE(TYPE_REF(tpp)))
                 tpp = TYPE_REF(tpp);
-            if (TYPE_REF(tpp) != NULL && !type_is_soft_compatible(tq, TYPE_REF(tpp)))
-                goto no_compatible;
-            if (TYPE_REF(tpp) == NULL ||
```

Comment:
Why do you remove this code (lines 2306-2307) ?

Reply:
because they are replaced by the one below